kythe/java/com/google/devtools/kythe/platform/kzip/KZipWriter.java

```diff
@@ -81,10 +88,14 @@ public String writeFile(byte[] data) throws IOException {
    * @param path path in the zip file to create
    */
   private void appendZip(byte[] data, String path) throws IOException {
-    ZipEntry entry = new ZipEntry(path);
-    output.putNextEntry(entry);
-    output.write(data);
-    output.closeEntry();
+    if (pathsWritten.add(path)) {
+      ZipEntry entry = new ZipEntry(path);
+      output.putNextEntry(entry);
+      output.write(data);
+      output.closeEntry();
+    } else {
+      System.err.println("Warning: Already wrote " + path + " to kzip.");
```

Comment:
maybe flogger if cody pulled that in (I forget)

Reply:
done